feature-Min-packets

gfurri20 opened the pull request.

gfurri20 committed:
Create PCPRegistrationPacket.java

gfurri20 committed:
Update PCPRegistrationPacket.java

Alessio789 committed:
Added PCPDisconnectionPacket

Alessio789 committed:
Added PCPDisconnectionPacket

gfurri20 committed:
Create PCP packets

Alessio789 committed:
Added PCPGroupUsersList and PCPRegHack

JacopoWolf committed:
refractoring and minor corrections

- added MAX_PACKET_LENGHT in IPCPpacket
- refractored PCPMsgToGroupPacket

gfurri20 committed:
Update PCPMsgUserToUserPacket.java

JacopoWolf committed:
MsgUserToUser rewrtited toBytes and added test

JacopoWolf committed:
added co-author to MsgUserToUser

Commit:
Added getters and setters to all packets classes

gfurri20 committed:
Create TESTs

Alessio789 committed:
Changed type of users ID from String to byte[]

JacopoWolf committed:
Updated IPCPpacket and created PCP static class

IPCPpacket
- packets now must implement byte[] header()
PCP
- class contains static util variables, like PORT
- static subclasses contain version specific variables

gfurri20 committed:
Added header() methods

JacopoWolf commented on PCP/src/main/java/PCP/packets/PCPErrorPacket.java:
```diff
@@ -32,6 +41,16 @@ public OpCode getOpCode()
+    {
+        byte[] buffer = new byte[2];
+        buffer[0] = OpCode.Error.getByte();
+        buffer[1] = this.errorCode.getByte();
```
the error code is the payload

JacopoWolf replied:
error code is the payload of the ErrorPackage

JacopoWolf commented on PCP/src/main/java/PCP/packets/PCPGroupUsersList.java:
```diff
@@ -0,0 +1,141 @@
+    public PCPGroupUsersList( int type, int listLenght, String jsonContent ) 
+    {
+        this.type = type;
+        this.listLenght = listLenght;
```
list lenght should be calculated and not stored as a variable. 

I suggest using an `ArrayList<String>` and then convert it.
Remember those APIs are meant to be an easy interface to work with.

JacopoWolf commented on PCP/src/main/java/PCP/packets/PCPMsgUserToUserPacket.java:
```diff
@@ -0,0 +1,141 @@
+        int NpacketsToSent = 
+                (
+                    this.message.length() / 
+                    (PCP.Min.MAX_PACKET_LENGHT - 5 - this.destinationAlias.length())
```
note: this line could be calculated using `header.lenght`

JacopoWolf commented on PCP/src/main/java/PCP/packets/PCPMsgUserToUserPacket.java:
```diff
@@ -0,0 +1,141 @@
+        for ( int packetN = 0; packetN < NpacketsToSent; packetN++ )
+        {
+        
+            byte[] buffer = this.header();
```
this will inevitably generate an **IndexOutOfBounds** exception, because the buffer lenght is the header's.
Should be corrected in other files if the error is repeated

gfurri20 replied:
No because header() returns a buffer which is initialize with the length take from size(). However is wrong because the header() method have to return a the header so the length must corrispond to the header bytes

JacopoWolf commented on PCP/src/main/java/PCP/packets/PCPRegistrationPacket.java:
```diff
@@ -0,0 +1,99 @@
+        for(byte b : topic.getBytes())
+            buffer[i++] = b;
+        //Delimitator
+        buffer[i++] = 0;
```
I would consides this as the payload

JacopoWolf commented on PCP/src/test/java/PCP/packets/PCPGroupUserList_TEST.java:
```diff
@@ -0,0 +1,42 @@
+            Assert.fail();
+        
+        Assert.assertArrayEquals( expectedResult, results.get(0) );
+    }
```
Based on the comment on PCPGroupUserListPacket this is wrong

JacopoWolf replied:
i will personally write this in a second moment

JacopoWolf commented on PCP/src/main/java/PCP/packets/PCPAliasChangePacket.java:
```diff
@@ -0,0 +1,112 @@
+        for(byte b : newAlias.getBytes())
+            buffer[i++] = b;
+        //Delimitator
+        buffer[i++] = 0;
```
old alias and new alias are this package's payload. Should go in toBytes()

JacopoWolf commented on PCP/src/main/java/PCP/packets/PCPDisconnectionPacket.java:
```diff
@@ -0,0 +1,112 @@
+public class PCPDisconnectionPacket implements IPCPpacket
+{
+    private byte[] id;
+    private int reason;
```
private variable could be a byte. conversion is going to be effetuated in the constructor and in the accessors.

gfurri20 replied:
I think that this piece of code i right:

- we can handle the id directly in byte[] form
- the reason will be converted in the constructor (or in the methods)

JacopoWolf replied:
I was thinking about `reason` but i guess you're right. 
We could also create an enum inside of DisconnectionPacket with the specified opcodes

gfurri20 replied:
Yes it would be even more orderly.

JacopoWolf commented:
And i still have to run the tests

gfurri20 commented:
I didn't understand what elements you thought were to be included in the payload. Now I got it

JacopoWolf commented:
Yeah sorry I've been a bit vague. thought it was obvius

JacopoWolf committed:
Renamed test classes to be run.

JacopoWolf committed:
Merge branch 'dev-Min' into feature-Min-packets

JacopoWolf committed:
Update PCPGroupUserList_Test.java

to be rewritten

JacopoWolf committed:
Merge branch 'feature-Min-packets' of https://github.com/JacopoWolf/PotatoChatProtocol into feature-Min-packets

JacopoWolf commented:
Yeah had a moment of confusion. But solved a merge conflict.

I deleted a test to be rewritten later, because of the GroupList error said above

gfurri20 committed:
Changes to header methods

JacopoWolf committed:
Update PCPDisconnectionPacket.java

added PCPDisconnectionpacket.Reason and consequent modifications

JacopoWolf commented:
As of now, the only things left to fix are the orrendous `PCPGroupUserListPacket` and issue #17 .
Then I will re-review the code and approve it

JacopoWolf committed:
Changed message packets hierarchy

JacopoWolf committed:
Fixed critical message packet creation mess

JacopoWolf commented:
Well i also enormously simplified packet creation by generically implementing `toBytes()` in the abstract class `PCPMessage` with my last two commits.

gfurri20 committed:
Added dependency gson and fixed toBytes() method for PCPGroupUsersList

gfurri20 committed:
Update PCPGroupUsersList.java

gfurri20 committed:
Update PCPGroupUsersList.java

gfurri20 committed:
Update PCPGroupUsersList.java

JacopoWolf commented on PCP/src/main/java/PCP/packets/PCPGroupUsersList.java:
```diff
@@ -0,0 +1,168 @@
+    private int type;
+    private int listLenght;
+    private ArrayList<String> listOfUsers;
+    private String jsonContent;
```
jsoncontent is supposed to be **calculated** with a method, and not be a variable.

JacopoWolf committed:
Code enhancment

- rewrote group user list packet
- renamed classes to keep continuity in naming
- moved variable payload formation to a specialized abstract class
